### PR TITLE
feat : Added basic auth with test credentials

### PR DESCRIPTION
**Pull request by rutvikpumak**
->User not able to go to the cart page, wishlist page, user profile page without login.
-> For now, log in via only hardcoded test credentials.
->Navigate based on conditional routing.
->Added user profile page.
->Log out feature added for user.
->Try to play with it and let me know if you find any bugs.

Preview : https://deploy-preview-2--confident-montalcini-5b49e9.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+import {
+  Cart,
+  Home,
+  Product,
+  ProductPage,
+  UserProfile,
+  Wishlist,
+} from "./pages/index.js";
 import Mockman from "mockman-js";
 import Navbar from "./component/Navbar/Navbar";
```
Comment (znjs) [inline src/App.js:12]: Suggestion: Can use barrel import

File: src/component/Navbar/Navbar.js [hunk 2]
```diff
@@ -29,41 +32,44 @@ export default function Navbar() {
...
+              className="nav-cart"
+              onClick={() => (token ? navigate("/cart") : navigate("/login"))}
+            >
+              <div className="icon cart-badge">
+                <i className="fa fa-shopping-cart "></i>
+                {/* {cartCount > 0 && (
                       <div className="notification-icon flex-center">
                         <span>{cartCount}</span>
                       </div>
                     )} */}
```
Comment (anuj-kumary) [inline src/component/Navbar/Navbar.js]: Remove unwanted comments

File: src/component/Navbar/Navbar.js [hunk 3]
```diff
@@ -1,7 +1,10 @@
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../context/auth/authContext";
```
Comment (znjs) [inline src/component/Navbar/Navbar.js]: Suggestion: Can use barrel import

File: src/context/auth/authContext.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { createContext, useContext, useState } from "react";
+import { loginService } from "../../services/login/loginService";
```
Comment (znjs) [inline src/context/auth/authContext.js]: Suggestion: Can use barrel import

File: src/context/index.js [hunk 5]
```diff
@@ -1 +1,3 @@
 export { useData, DataProvider } from "./data/dataContext";
+
+export { useAuth, AuthProvider } from "./auth/authContext";
```
Comment (znjs) [inline src/context/index.js:3]: Suggestion: Can use barrel import

File: src/pages/Product/ProductPage.js [hunk 6]
```diff
@@ -2,14 +2,13 @@ import React from "react";
 import "./ProductPage.css";
 import { useNavigate, useParams } from "react-router";
 import { calcPercentage } from "./components/ProductCard";
```
Comment (znjs) [inline src/pages/Product/ProductPage.js:4]: Suggestion: Can use barrel import

**Review by znjs — APPROVED** [review]
Clean code
Good File Structure
Good Naming conventions used

File: src/pages/Auth/Login.js [hunk 7]
```diff
@@ -1,57 +1,96 @@
...
 export function Login() {
+  const [loginForm, setLoginForm] = useState({
+    email: "",
+    password: "",
+  });
+  const navigate = useNavigate();
+  const { token, loginUser } = useAuth();
+
+  useEffect(() => {
+    loginUser(loginForm.email, loginForm.password);
```
Comment (abhisheksahani-nova) [inline src/pages/Auth/Login.js]: Use async await here too

Comment (rutvikpumak) [reply src/pages/Auth/Login.js]: updated , please check now

File: src/pages/Product/ProductPage.js [hunk 8]
```diff
@@ -2,14 +2,13 @@ import React from "react";
...
-import { useData } from "../../context";
+import { useAuth, useData } from "../../context";
 
 export function ProductPage() {
   const { productId } = useParams();
   const navigate = useNavigate();
-
+  const { token } = useAuth();
   const { products } = useData();
-  console.log(products, productId);
```
Comment (abhisheksahani-nova) [inline src/pages/Product/ProductPage.js:12]: Remove console.log from the code, once used and tested

**Review by abhisheksahani-nova — APPROVED** [review]
Suggestion :- 

1. use token directly as key and value instead of storing in object
2. Remove comments if not needed.


Else evrything looks subtle , good to merge , great work

**Review by Mandhularajitha — COMMENTED** [review]
margin,Clean code, File Structure.

**Review by Mandhularajitha — COMMENTED** [review]
File Structure ,and margin is  good, Clean code

**Pull request merged**